### PR TITLE
Fix: stream end when length of last chunk equal to MAX_CHUNK

### PR DESCRIPTION
**Pull request by PrimozGodec**
This pull request is fixing bug https://github.com/Lukasa/hyper/issues/355

I changed the approach of taking chunks form the generator, such that we know when we reach the last chunk with memorising the next chunk.

### CODE REVIEW

**Review by Lukasa — CHANGES_REQUESTED** [review]
Yup, this looks like it’ll work. Mind adding an entry to the change log and also a test for me?

**Comment by PrimozGodec** [comment]
@Lukasa I changed the code and made unit test pass on my machine. I also added a test to test specific change. 
Sorry I can not find a change log. Can you tell me where is it?
Also travis tests are failing it seems that they are failing because of some older code and not because of my changes.

**Comment by matjazp** [comment]
@PrimozGodec, CI is fixed now, can you rebase please?

Also, change log is in HISTORY.rst and CONTRIBUTORS.rst.

**Comment by PrimozGodec** [comment]
Thank you @matjazp. It is fixed now @Lukasa

File: hyper/http20/stream.py [hunk 1]
```diff
@@ -122,8 +122,17 @@ def file_iterator(fobj):
             chunks = (data[i:i+MAX_CHUNK]
                       for i in range(0, len(data), MAX_CHUNK))
 
-        for chunk in chunks:
-            self._send_chunk(chunk, final)
+        # since we need to know when we have a last package we need to know
+        # if there is another package in advance
+        cur_chunk = next(chunks)
```
Comment (Lukasa) [inline hyper/http20/stream.py]: This will explode if there are no chunks to send: I think you need to write a test case for that and refactor.

**Review by Lukasa — CHANGES_REQUESTED** [review]

File: hyper/http20/stream.py (hunk 1 above)
Comment (PrimozGodec) [reply hyper/http20/stream.py]: Done

**Review by Lukasa — APPROVED** [review]
Looks good to me, thanks! :sparkles:

**Pull request merged**